albumentations/augmentations/transforms.py

```diff
@@ -105,14 +140,11 @@ class RandomGridShuffle(DualTransform):
     _targets = (Targets.IMAGE, Targets.MASK, Targets.KEYPOINTS)
 
     def __init__(self, grid: Tuple[int, int] = (3, 3), always_apply: bool = False, p: float = 0.5):
-        super().__init__(always_apply, p)
-
-        n, m = grid
-
-        if not all(isinstance(dim, int) and dim > 0 for dim in [n, m]):
-            raise ValueError(f"Grid dimensions must be positive integers. Current grid dimensions: [{n}, {m}]")
-
-        self.grid = grid
+        config = RandomGridShuffleConfig(grid=grid, always_apply=always_apply, p=p)
```

Comment:
Maybe we can do this using metcaclasses...
Look SerializableMeta as an example https://github.com/albumentations-team/albumentations/blob/main/albumentations/core/serialization.py#L44